core/proc/shared/dup.rb

```diff
@@ -7,4 +7,12 @@
 
     a.call.should == b.call
   end
+
+  ruby_version_is "3.2" do
+    it "returns an instance of subclass" do
+      cl = Class.new(Proc)
+
+      cl.new{}.dup.class.should == cl
```

Comment:
**major:** It's a shared example and used for both `#dup` and `#clone` methods. It's parametrised with `@method` instance variable, that may be either `:clone` or `:dup`.

So instead of hardcoding the `dup` method call (`cl.new{}.dup`) a proper method should be called with `send(@method)` (you may find a test case above useful).